### PR TITLE
Fix broken testcafe search tests

### PR DESCRIPTION
**Pull request by cdrini**
Re-enable + fix testcafe search tests for shadow dom.

- testcafe was erroring with `BRSearchInProgress` undefined
- Reaching into shadow DOM 😬 We probably want to setup a proper pattern for doing this, probably ok for now?
- dom.searchNavigation was null in github CI; but worked locally for me

### CODE REVIEW

File: src/plugins/search/plugin.search.js [hunk 1]
```diff
@@ -201,21 +197,14 @@ BookReader.prototype.search = function(term = '', overrides = {}) {
...
   };
 
   this.trigger('SearchStarted', { term: this.searchTerm, instance: this });
   return $.ajax({
     url: url,
     dataType: 'jsonp',
     cache: true,
-    beforeSend,
-    jsonpCallback: 'BRSearchInProgress'
+    beforeSend: xhr => { this.searchXHR = xhr; },
```
Comment (iisa) [inline src/plugins/search/plugin.search.js:207]: interesting, we don't need `jsopCallback` anymore?

File: tests/e2e/helpers/desktopSearch.js [hunk 2]
```diff
@@ -19,14 +19,15 @@ export function runDesktopSearchTests(br) {
...
+      await t.click(nav.desktop.searchIcon);
 
       //testing search for a word found in the book
-      await t
-        .selectText(nav.desktop.searchBox.child('.BRsearchInput'))
-        .pressKey('delete');
-      await t.typeText(nav.desktop.searchBox.child('.BRsearchInput'), TEST_TEXT_FOUND);
-      await t.click((nav.desktop.searchBox).child('.BRsearchSubmit'));
+      await t.selectText(nav.desktop.searchBox).pressKey('delete');
+      // FIXME: Why is it only typing every other letter?!?!
```
Comment (iisa) [inline tests/e2e/helpers/desktopSearch.js:27]: whoa

File: tests/e2e/models/Navigation.js [hunk 3]
```diff
@@ -35,7 +40,11 @@ class DesktopNav {
     this.zoomOut = bottomToolbar.find('.BRicon.zoom_out');
 
     // search
-    this.searchBox = topToolbar.find('.BRbooksearch.desktop');
+    this.searchIcon = itemNav.find('button.shortcut.search');
+    this.searchBox = itemNav
```
Comment (iisa) [inline tests/e2e/models/Navigation.js:44]: yay

File: src/plugins/search/plugin.search.js (hunk 1 above)
Comment (cdrini) [reply src/plugins/search/plugin.search.js:207]: Testcafe was giving a weird error about `BRSearchInProgress` not being defined? So we're now using `$.ajax`'s implicit jsonp callback. Should behave the same! (Might cause caching issues now I think about it maybe?)

File: src/plugins/search/view.js [hunk 4]
```diff
@@ -149,10 +149,12 @@ class SearchView {
   }
 
   updateResultsPosition() {
+    if (!this.dom.searchNavigation) return;
     this.dom.searchNavigation.find('[data-id=resultsCount]').text(this.resultsPosition());
   }
 
   updateSearchNavigationButtons() {
+    if (!this.dom.searchNavigation) return;
```
[the remark this answers is not in the document]
Comment (cdrini) [reply src/plugins/search/view.js:157]: Not sure :/ Testcafe was failing here on GH. But it was running fine locally :/

**Review by iisa — APPROVED** [review]

**Pull request merged**